staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go

```diff
@@ -763,45 +768,51 @@ spec:
 		t.Fatalf("unexpected create error: %v", err)
 	}
 
-	// wait for condition with violations
-	t.Log("Waiting for NonStructuralSchema condition")
-	var cond *apiextensionsv1beta1.CustomResourceDefinitionCondition
-	err = wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (bool, error) {
-		obj, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), name, metav1.GetOptions{})
+	testExpectedViolations := func(t *testing.T) {
+		// wait for condition with violations
+		t.Log("Waiting for NonStructuralSchema condition")
+		err = wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (bool, error) {
+			obj, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			cond = findCRDCondition(obj, apiextensionsv1beta1.NonStructuralSchema)
+			return cond != nil, nil
+		})
 		if err != nil {
-			return false, err
+			t.Fatalf("unexpected error waiting for NonStructuralSchema condition: %v", cond)
+		}
+		for _, violation := range expectedViolations {
+			if !strings.Contains(cond.Message, violation) {
+				t.Fatalf("expected violation %q, but got: %v", violation, cond.Message)
+			}
```

Comment:
am still a little unsure and surprised about this big fallout here in the test. Would expect something very local. Isn't there a way to rewrite this test by adding steps only without rewriting what we have?

Reply:
I agree the change is big and the same can be achieved by also adding the steps. But the intent was to just refactor the code to avoid duplication. This part of code does not change any of the previous logic and only adds some checking for `preserveUnknownFields `

Reply:
made the changes to update the steps locally